Relationships/RelationshipResolver.cs

```diff
@@ -123,22 +145,22 @@ private void Resolve(CkanModule module, RelationshipResolverOptions options)
             // Even though we may resolve top-level suggests for our module,
             // we don't install suggestions all the down unless with_all_suggests
             // is true.
-            var sub_options = (RelationshipResolverOptions)options.Clone();
+            var sub_options = (RelationshipResolverOptions) options.Clone();
             sub_options.with_suggests = false;
 
             log.DebugFormat("Resolving dependencies for {0}", module.identifier);
-            ResolveStanza(module.depends, sub_options);
+            ResolveStanza(module.depends, new Relationship.Depends(module), sub_options);
 
             if (options.with_recommends)
             {
                 log.DebugFormat("Resolving recommends for {0}", module.identifier);
-                ResolveStanza(module.recommends, sub_options, true);
+                ResolveStanza(module.recommends, new Relationship.Recommended(module), sub_options, true);
```

Comment:
Noting to self: if we're processing recommends/suggests, we're doing a 'soft' resolve. These can fail without breaking the whole process.